doc/devel/api_changes.rst

```diff
@@ -137,6 +143,30 @@ Expire deprecation
      instead. For removed items that were not in the stub file, only deleting from the
      allowlist is required.
 
+.. _pending-deprecation:
+
+Pending deprecations
+^^^^^^^^^^^^^^^^^^^^
+
+If the deprecation is introduced as part of implementing an alternative, then the
+deprecation is marked as pending and introduced one meso release after the feature.
```

Comment:
```suggestion
Pending are an announcement that a deprecation will happen in the future. By default, pending deprecations
do not raise a warning to the user. They are primarily intended as a heads-up to give time to adapt before
a deprecation warning will be issued. This is in particular useful for downstream libraries. They can adapt
before their code raises a warning. This is a service to users of the downstream libraries, because they cannot
act on the warning and thus should ideally not be bothered by it.

Pending deprecations still show up in the documentation and release notes. It's also possible to run
Python in dev mode to raise `PendingDeprecationWarning`.
```